garrysmod/lua/includes/extensions/player.lua

```diff
@@ -319,11 +330,30 @@ function player.Iterator()
 
 end
 
-local function InvalidatePlayerCache( ent )
-
-	if ( ent:IsPlayer() ) then PlayerCache = nil end
-
+hook.Add( "OnEntityCreated", "PlayerCache", function( ent )
+	if ( ent:IsPlayer() ) then 
+		PlayerCache = nil
+		timer.Create("PlayerCache",0,1,function()

```

Comment:
So you wait for the next frame before caching the variables. What happens if those functions (`player.GetBySteamID`, etc.) are called during this time? 

That smells like a regression.

Is this timer even necessary? From my brief test the SteamID is already set by this point.

Reply:
one of the functions has to be delayed a frame because of https://github.com/Facepunch/garrysmod-issues/issues/6012